storage-proofs-porep/src/stacked/vanilla/cache.rs

```diff
@@ -155,11 +157,26 @@ impl ParentCache {
         G: Graph<H> + ParameterSetMetadata + Send + Sync,
     {
         let path = cache_path(cache_entries, graph);
+        let generation_key = path.display().to_string();
+        let mut generated = PARENT_CACHE_GENERATION_LOCK
+            .lock()
+            .expect("parent cache generation lock failed");
+
         if path.exists() {
+            // If the cache file exists and we've got the lock, generation has previously been
+            // completed.  Insert that it no longer needs generation at this point unconditionally.
+            if generated.get(&generation_key).is_none() {
+                generated.insert(generation_key);
+            }
             Self::open(len, cache_entries, graph, &path)
         } else {
             match Self::generate(len, cache_entries, graph, &path) {
-                Ok(c) => Ok(c),
+                Ok(c) => {
+                    generated.insert(generation_key);
+                    drop(generated);
```

Comment:
this is not really needed, as it will be dropped as soon as the function ends

Reply:
I thought an explicit drop made sense at the time, but agreed, it's not needed